Lib/test/test_listcomps.py

```diff
@@ -762,6 +762,18 @@ def g():
         """
         self._check_in_scopes(code, {"x": 1, "y": 1}, scopes=["module", "function"])
 
+    def test_freevar_through_scope_containing_comprehension_with_cell(self):
+        code = """
+            x = 1
+            def f():
+                [lambda: x for x in [1]]
```

Comment:
```suggestion
                [lambda: x for x in [2]]
```
As discussed:
- Use a different value (the test ran correctly but used the wrong `1`)
- Should change it to execute the lambda so we ensure the value of `x` inside the lambda is right